eng/Versions.props

```diff
@@ -12,40 +12,44 @@
-    <MicrosoftExtensionsLoggingVersion>8.0.0</MicrosoftExtensionsLoggingVersion>
-    <MicrosoftExtensionsObjectPoolVersion>8.0.7</MicrosoftExtensionsObjectPoolVersion>
-    <MicrosoftExtensionsOptionsVersion>8.0.2</MicrosoftExtensionsOptionsVersion>
+    <MicrosoftBclAsyncInterfacesVersion>9.0.4</MicrosoftBclAsyncInterfacesVersion>
+    <MicrosoftExtensionsDependencyInjectionVersion>9.0.4</MicrosoftExtensionsDependencyInjectionVersion>
+    <MicrosoftExtensionsDependencyInjectionAbstractionsVersion>9.0.4</MicrosoftExtensionsDependencyInjectionAbstractionsVersion>
+    <MicrosoftExtensionsLoggingVersion>9.0.4</MicrosoftExtensionsLoggingVersion>
+    <MicrosoftExtensionsObjectPoolVersion>9.0.4</MicrosoftExtensionsObjectPoolVersion>
+    <MicrosoftExtensionsOptionsVersion>9.0.4</MicrosoftExtensionsOptionsVersion>
     <NuGetVersion>6.9.1</NuGetVersion>
     <SkiaSharpVersion>2.88.8</SkiaSharpVersion>
-    <SystemCodeDomVersion>8.0.0</SystemCodeDomVersion>
-    <SystemCollectionsImmutableVersion>8.0.0</SystemCollectionsImmutableVersion>
-    <SystemConfigurationConfigurationManagerVersion>6.0.1</SystemConfigurationConfigurationManagerVersion>
-    <SystemFormatsAsn1Version>8.0.1</SystemFormatsAsn1Version>
+    <SystemCodeDomVersion>9.0.4</SystemCodeDomVersion>
+    <SystemCollectionsImmutableVersion>9.0.4</SystemCollectionsImmutableVersion>
+    <SystemConfigurationConfigurationManagerVersion>9.0.4</SystemConfigurationConfigurationManagerVersion>
+    <SystemFormatsAsn1Version>9.0.4</SystemFormatsAsn1Version>
     <SystemIOFileSystemAccessControl>5.0.0</SystemIOFileSystemAccessControl>
-    <SystemNumericsTensorsVersion>9.0.0</SystemNumericsTensorsVersion>
+    <SystemIOPipelinesVersion>9.0.4</SystemIOPipelinesVersion>
+    <SystemNumericsTensorsVersion>9.0.4</SystemNumericsTensorsVersion>
     <SystemReflectionEmitLightweightVersion>4.7.0</SystemReflectionEmitLightweightVersion>
-    <SystemReflectionEmitVersion>4.3.0</SystemReflectionEmitVersion>
+    <SystemReflectionEmitVersion>4.7.0</SystemReflectionEmitVersion>
     <SystemSecurityPrincipalWindows>5.0.0</SystemSecurityPrincipalWindows>
-    <SystemTextEncodingsWebVersion>8.0.0</SystemTextEncodingsWebVersion>
-    <SystemTextJsonVersion>8.0.5</SystemTextJsonVersion>
-    <SystemThreadingChannelsVersion>8.0.0</SystemThreadingChannelsVersion>
+    <SystemTextEncodingsWebVersion>9.0.4</SystemTextEncodingsWebVersion>
+    <SystemTextJsonVersion>9.0.4</SystemTextJsonVersion>
+    <SystemThreadingChannelsVersion>9.0.4</SystemThreadingChannelsVersion>
     <!-- Other product dependencies -->
     <ApacheArrowVersion>14.0.2</ApacheArrowVersion>
-    <GoogleProtobufVersion>3.27.1</GoogleProtobufVersion>
+    <GoogleProtobufVersion>3.30.2</GoogleProtobufVersion>
     <LightGBMVersion>4.6.0</LightGBMVersion>
+    <JsonSchemaNetGenerationVersion>5.0.1</JsonSchemaNetGenerationVersion>
+    <JsonSchemaNetVersion>7.3.4</JsonSchemaNetVersion>
```

Comment:
What these used for? I mean why did we add these libraries dependency? These looks third party libraries, right?

Reply:
They were always dependencies of dependencies. I added them in directly here so we could get the latest version and fix the nuget warnings. Would another way be better?

Reply:
I am not sure this can cause issue the direct dependency expects lower version of these added references. @ericstj may advise about that

Reply:
We can use CPM in a follow up PR.  Upgrading the versions of transitive dependencies is fine, and we'll get a warning if we are ever behind.  Listing everything gives dependabot better chance at keeping things up to date.